code/modules/surgery/organs/blood.dm

```diff
@@ -1,87 +1,37 @@
 
-	vessel = new/datum/reagents(max_blood)
-	vessel.my_atom = src
 
-	if(species && species.exotic_blood)
-		vessel.add_reagent(species.exotic_blood, max_blood)
-	else
-		vessel.add_reagent("blood", max_blood)
-		for(var/datum/reagent/blood/B in vessel.reagent_list)
-			if(B.id == "blood")
-				B.data = list(
-				"donor" = src,
-				"viruses" = null,
-				"blood_DNA" = dna.unique_enzymes,
-				"blood_colour" = species.blood_color,
-				"blood_type" = b_type,
-				"resistances" = null,
-				"trace_chem" = null)
-	spawn(1)
-		fixblood()
-
-//Resets blood data
-/mob/living/carbon/human/proc/fixblood()
-	for(var/datum/reagent/blood/B in vessel.reagent_list)
-		if(B.id == "blood")
-			B.data = list(
-			"donor" = src,
-			"viruses" = null,
-			"blood_DNA" = dna.unique_enzymes,
-			"blood_colour" = species.blood_color,
-			"blood_type" = b_type,
-			"resistances" = null,
-			"trace_chem" = null)
 
 // Takes care blood loss and regeneration
 /mob/living/carbon/human/proc/handle_blood()
-	var/blood_volume
+	var/list/blood_data = get_blood_data(get_blood_id())//PROCCEPTION
 	if(species && species.flags & NO_BLOOD)
+		bleed_rate = 0
 		return
-	if(stat != DEAD && bodytemperature >= 170)	//Dead or cryosleep people do not pump the blood.
-		if(species.exotic_blood)
-			var/blood_reagent = species.exotic_blood // This is a string of the name of the species' blood reagent
-			blood_volume = round(vessel.get_reagent_amount(blood_reagent))
-			if(blood_volume < max_blood && blood_volume)
-				vessel.add_reagent(blood_reagent, 0.1) // regenerate blood VERY slowly
-				if(reagents.has_reagent(blood_reagent))
-					vessel.add_reagent(blood_reagent, 0.4)
 
-		else
-			blood_volume = round(vessel.get_reagent_amount("blood"))
-			//Blood regeneration if there is some space
-			if(blood_volume < max_blood && blood_volume)
-				var/datum/reagent/blood/B = locate() in vessel.reagent_list //Grab some blood
-				if(B) // Make sure there's some blood at all
-					if(mind) //Handles vampires "eating" blood that isn't their own.
-						if(mind in ticker.mode.vampires)
-							for(var/datum/reagent/blood/BL in vessel.reagent_list)
-								if(nutrition >= NUTRITION_LEVEL_WELL_FED)
-									break //We don't want blood tranfusions making vampires fat.
-								if(BL.data["donor"] != src)
-									nutrition += (15 * REAGENTS_METABOLISM)
-									BL.volume -= REAGENTS_METABOLISM
-									if(BL.volume <= 0)
-										qdel(BL)
-									break //Only process one blood per tick, to maintain the same metabolism as nutriment for non-vampires.
-
-					if(B.data["donor"] != src) //If it's not theirs, then we look for theirs
-						for(var/datum/reagent/blood/D in vessel.reagent_list)
-							if(D.data["donor"] == src)
-								B = D
-								break
-
-					vessel.add_reagent("blood", 0.1) // regenerate blood VERY slowly
+	if(stat != DEAD && bodytemperature >= 225)	//Dead or cryosleep people do not pump the blood.
```

Comment:
`if(bodytemperature >= 225 && !( NOCLONE in mutations)) //cryosleep or husked people do not pump the blood.`

The dead check shoudln't be necessary; this portion of `Life` shouldn't run if they're dead; if it is, that's a larger bug.